src/javascripts/ng-admin/Crud/field/maReferenceField.js

```diff
@@ -0,0 +1,87 @@
+function maReferenceField($compile, ReadQueries) {
+    return {
+        scope: {
+            'field': '&',
+            'value': '=',
+            'entry':  '=?',
+            'datastore': '&?'
+        },
+        restrict: 'E',
+        compile: function() {
+            return {
+                pre: function(scope, element) {
```

Comment:
I didn't pay attention when you updated `<ma-choice-field>`, but why do you need this instead of a simple `link` function?